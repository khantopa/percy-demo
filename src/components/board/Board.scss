@import '../../styles/main.scss';

.play {
  width: 100%;
  height: 100vh;
  padding: 20px 40px;
  box-sizing: border-box;
  &_title {
    text-align: center;
    color: $light;
  }
  &_board {
    position: relative;
    padding: 20px 40px;
    height: 300px;  
    &_players {
      @include grid(2);
      color: $light;
      margin-bottom: 20px;
    }
    &_result {
      text-align: center;
      animation: fadeIn .5s ease;
      color: $light;
      font-size: 20px;
      font-weight: bolder;
    }
  }
  &_button { 
    width: 300px;
    margin: 10px auto;
    @include grid(3);
    grid-gap: 10px;
    &_item {
     @extend .icon_button;
    }
  }
  &_text {
    text-align: center;
    color: black;
    font-size: bolder;
  }
  &_run {
    @include grid(1);
    &_button {
      @extend .icon_button;
    }
  }
  &_footer {
    width: 100%;
    margin-top: 20px; 
    box-sizing: border-box;
    @include grid(2);
    grid-gap: 20px;
    &_button {
      @extend .button;
      width: 200px;
      background: $dark;
      &.reset {
        background: $secondary;
      }
    }
  }
}

@media (max-width: $mobile-width) {
  .play {
    width: 100vw;
    padding: 2px;
    &_board {
      padding: 20px 0;
      height: 230px;
    }
    &_footer {
      margin-top: 40px;
      @include grid(1);
      grid-gap: 10px;
    }
  }
}