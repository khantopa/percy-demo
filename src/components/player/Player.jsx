import React from 'react';
import PropTypes from 'prop-types';

import './Player.scss';

import Symbol from '../symbol/Symbol';


const Player = (props) => {

  const player = props.player;
  return (
    <div className="player">
      <h5 className="player_type">
        { player.name }: &nbsp;
        <span className="player_score"> { player.score }  </span>
      </h5>
      <Symbol symbol= { player.symbol } />
    </div>
  );
}

Player.propTypes = {
  player: PropTypes.object.isRequired
}

export default Player;

