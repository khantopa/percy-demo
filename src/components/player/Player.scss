@import '../../styles/main.scss';

.player {
  height: 200px;
  width: 200px;
  animation: fadeIn 2s ease;
  &_type {
    text-align: center;
    font-size: 16px;
  }
}

@media (max-width: $mobile-width) {
  .player {
    width: 130px;
    height: 130px;
    &_type {
      font-size: 14px;
    }
  }
}