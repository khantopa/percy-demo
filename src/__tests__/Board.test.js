import Board from '../components/Board';
import Player from '../components/Player';


let wrapper, props, state, runButton, player1, player2, winner;

beforeEach(() => {
  state = {
    mode: "Player vs Computer"
  }
  props = { state }
  wrapper = shallow(<Board location={props} />);
});

describe('<Board>', () => {

  it('should <Board> defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('should <Board> defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders <Board> without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render <Player>', () => {
    expect(wrapper.find(Player)).toHaveLength(2);
  });

  it('should call fetch when mounted', () => {
    expect(wrapper.instance().UNSAFE_componentWillMount());
  });

});


describe('Computer vs Computer', () => {

  it('Random vs Random', () => {
    jest.useFakeTimers();
    state.mode = 'Computer vs Computer';
    wrapper = shallow(<Board location={props} />)
    wrapper.instance().UNSAFE_componentWillMount();
    runButton = wrapper.find('button.play_run_button');
    runButton.simulate('click');
    jest.advanceTimersByTime(2000);
    player1 = wrapper.state('player1');
    player2 = wrapper.state('player2');
    winner = wrapper.state('winner');
    if (player1.symbol === player2.symbol) {
      expect(winner).toBe('Draw !')
    } else if ((player1.symbol === "rock" && player2.symbol === "scissors") ||
      (player1.symbol === "paper" && player2.symbol === "rock") ||
      (player1.symbol === "scissors" && player2.symbol === "paper")) {
        expect(winner).toBe(`${player1.name} wins !`);
        expect(player1.score).toBe(1)
    } else {
      expect(winner).toBe(`${player2.name} wins !`);
      expect(player2.score).toBe(1)
    }
  });

});

describe('Player vs Computer', () => {

  it('Rock vs Random', () => {
    jest.useFakeTimers();
    state.mode = 'Player vs Computer';
    wrapper.instance().UNSAFE_componentWillMount();
    runButton = wrapper.find('button.play_button_item').first();
    runButton.simulate('click');
    jest.advanceTimersByTime(2000);
    player1 = wrapper.state('player1');
    player2 = wrapper.state('player2');
    winner = wrapper.state('winner');
    if (player1.symbol === player2.symbol) {
      expect(winner).toBe('Draw !')
    } else if (player2.symbol === "scissors") {
        expect(winner).toBe(`${player1.name} wins !`);
        expect(player1.score).toBe(1)
    } else {
      expect(winner).toBe(`${player2.name} wins !`);
      expect(player2.score).toBe(1)
    } 
  });

  it('Paper vs Random', () => {
    jest.useFakeTimers();
    state.mode = 'Player vs Computer';
    wrapper.instance().UNSAFE_componentWillMount();
    runButton = wrapper.find('button.play_button_item').at(1);
    runButton.simulate('click');
    jest.advanceTimersByTime(2000);
    player1 = wrapper.state('player1');
    player2 = wrapper.state('player2');
    winner = wrapper.state('winner');
    if (player1.symbol === player2.symbol) {
      expect(winner).toBe('Draw !')
    } else if (player2.symbol === "rock") {
        expect(winner).toBe(`${player1.name} wins !`);
        expect(player1.score).toBe(1)
    } else {
      expect(winner).toBe(`${player2.name} wins !`);
      expect(player2.score).toBe(1)
    } 
  });

  it('Scissor vs Random', () => {
    jest.useFakeTimers();
    state.mode = 'Player vs Computer';
    wrapper.instance().UNSAFE_componentWillMount();
    runButton = wrapper.find('button.play_button_item').at(2);
    runButton.simulate('click');
    jest.advanceTimersByTime(2000);
    player1 = wrapper.state('player1');
    player2 = wrapper.state('player2');
    winner = wrapper.state('winner');
    if (player1.symbol === player2.symbol) {
      expect(winner).toBe('Draw !')
    } else if (player2.symbol === "paper") {
        expect(winner).toBe(`${player1.name} wins !`);
        expect(player1.score).toBe(1)
    } else {
      expect(winner).toBe(`${player2.name} wins !`);
      expect(player2.score).toBe(1)
    } 
  });

});

