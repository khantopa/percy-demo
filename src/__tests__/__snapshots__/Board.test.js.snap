// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board> renders <Board> without crashing 1`] = `
<div
  className="play"
>
  <h3
    className="play_title"
  >
    Player vs Computer
  </h3>
  <div
    className="play_board"
  >
    <div
      className="play_board_players"
    >
      <Player
        player={
          Object {
            "name": "User",
            "score": 0,
            "symbol": "",
            "type": "user",
          }
        }
      />
      <Player
        player={
          Object {
            "name": "Computer",
            "score": 0,
            "symbol": "",
            "type": "computer",
          }
        }
      />
    </div>
    <br />
    <br />
    <div
      className="play_board_result"
    >
      <p
        className="play_board_result_text"
      >
        Start a new Game
      </p>
    </div>
  </div>
  <p
    className="play_text"
  >
    Choose any to play:
  </p>
  <div
    className="play_button"
  >
    <button
      className="play_button_item "
      disabled={false}
      key="rock"
      onClick={[Function]}
    >
      <Symbol
        color="#FFFEFC"
        symbol="rock"
      />
    </button>
    <button
      className="play_button_item "
      disabled={false}
      key="paper"
      onClick={[Function]}
    >
      <Symbol
        color="#FFFEFC"
        symbol="paper"
      />
    </button>
    <button
      className="play_button_item "
      disabled={false}
      key="scissors"
      onClick={[Function]}
    >
      <Symbol
        color="#FFFEFC"
        symbol="scissors"
      />
    </button>
  </div>
  <div
    className="play_footer"
  >
    <Link
      className="play_footer_button change_mode"
      to="/"
    >
      CHANGE MODE
    </Link>
    <button
      className="play_footer_button reset"
      onClick={[Function]}
    >
      RESET
    </button>
  </div>
</div>
`;
